src/org/openj9/envInfo/MachineInfo.java

```diff
@@ -104,17 +117,69 @@ private String execCommands(String[] commands) {
 		return rt;
 	}
 
+	private String parseInfo(String req) {
+		Pattern pattern = Pattern.compile("[0-9]+[.][0-9]+([.][0-9]+)?"); 
+		Matcher matcher = pattern.matcher(req);
+		if (matcher.find()) {
+			return matcher.group(0);
+		} else {
+			return "";
+		}
+	}
+
+	int[] versionStr2Arr(String ver) throws NumberFormatException {
+		String[] verSplit = ver.split("\\.");
+		int[] verArr = new int[verSplit.length];
+		for (int i=0; i < verSplit.length; i++) {
+			verArr[i] = Integer.parseInt(verSplit[i]);
+		}
+		return verArr;
+	}
+
+	private boolean validateVersion(String versionName, String actualVersionStr, String requriedVersionStr) {
+		boolean isValid = true;
+		try { 
+			int[] accVer = versionStr2Arr(actualVersionStr);
+			int[] reqVer = versionStr2Arr(requriedVersionStr);
```

Comment:
We need to handle the case that if accVer and reqVer has different length. One way to do it is to check if accVer.length == reqVer.length. If one is shorter than the other, add 0 in the shorter array. For example if accVer=1.1 reqVer=1.1.3, then append 0 in accVer which makes it 1.1.0.